backend/tests/data_products_datasets/test_router.py

```diff
@@ -48,6 +68,35 @@ def test_approve_data_product_link(
         approved_link = self.approve_default_data_product_dataset_link(client, link_id)
         assert approved_link.status_code == 200
 
+    def test_approve_data_product_link_by_admin(
```

Comment:
I think this tests the use case I described above? But then I don't know what the other use case does :D